pkg/querier/tripperware/shard_by_query.go

```diff
@@ -0,0 +1,81 @@
+package tripperware
+
+import (
+	"context"
+
+	"github.com/go-kit/log"
+	"github.com/thanos-io/thanos/pkg/querysharding"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+
+	cquerysharding "github.com/cortexproject/cortex/pkg/querysharding"
+)
+
+func ShardByMiddleware(logger log.Logger, limits Limits, merger Merger, numShards int) Middleware {
+	return MiddlewareFunc(func(next Handler) Handler {
+		return shardQueryByMiddleware{
+			next:      next,
+			limits:    limits,
+			merger:    merger,
+			logger:    logger,
+			numShards: numShards,
+		}
+	})
+}
+
+type shardQueryByMiddleware struct {
```

Comment:
should this be `shardByQueryMiddleware` to be consistent with the file name? S

Reply:
I renamed to `shard_by.go` only.. as this seems to be the pattern on the others middlewares.

WDYT?